src/main/java/com/clevertap/maven/plugins/supertest/SuperTestMavenPlugin.java

```diff
@@ -121,16 +149,88 @@ private StringBuilder buildProcessedMvnTestOpts(String artifactId, String groupI
     public int runShellCommand(final String command, final String commandDescriptor)
             throws IOException, InterruptedException {
         getLog().info("Running " + command);
-        Process proc = Runtime.getRuntime().exec(command);
+        ProcessBuilder pb = new ProcessBuilder(getShellCommandAsArray(command));
+        pb.redirectErrorStream(true);
+        Process proc = pb.start();
+        readProcessStdOut(proc, commandDescriptor);
+
+        // we don't want to wait forever, if something breaks
+        boolean exited = proc.waitFor(STDOUT_POST_READ_WAIT_TIMEOUT, TimeUnit.SECONDS);
+
+        if (exited) {
+            return proc.exitValue();
+        } else {
+            return 1;
```

Comment:
Can we kill the process instead of leaving it dangling?

Reply:
done